handlers/create_user.go

```diff
@@ -0,0 +1,46 @@
+package handlers
+
+import (
+	"encoding/json"
+	"http-go/models"
+	"http-go/storage"
+	"io/ioutil"
+	"net/http"
+)
+
+type adder interface {
+	Add(user *models.User) error
+}
+
+func CreateUser(src storage.Adder) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setupResponse(&w)
+
+		if r.Method != http.MethodPost {
+			http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+			return
+		}
+
+		user := models.User{}
```

Comment:
You can use `user := new(models.User)` then you won't need to pass it as a pointer to `src.Add(&user)` and can do just `src.Add(user)`. Same goes for `json.Unmarshal(data, &user)` you can do `json.Unmarshal(data, user)` instead.